src/main/java/org/jabref/gui/StateManager.java

```diff
@@ -128,6 +131,18 @@ public void setSearchQuery(SearchQuery searchQuery) {
         activeSearchQuery.setValue(Optional.of(searchQuery));
     }
 
+    public void setGlobalSearchActive(boolean active) {
```

Comment:
Do you really need to have this as a state variable? It seems like the only place where you need to know if the search bar is open already open is if the user starts a new search, so this seems to be a very localized thing.